### PR TITLE
test: add smoke test for settings menu

### PR DESCRIPTION
**Pull request by MarionLemaire**
### What does it do?

Adds smoke test for Settings menu.

Adds an enhancement to navToHeader to support looking for a nav item underneath a specific nav header

### Why is it needed?

To help me getting better at writing E2E tests !

### CODE REVIEW

File: e2e/tests/settings/smoke-test.spec.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+test.describe('Settings', () => {
+  test.beforeEach(async ({ page }) => {
+    await resetDatabaseAndImportDataFromPath('./e2e/data/with-admin.tar');
+    await page.goto('/admin');
+    await login({ page });
+  });
+
+  test('every expected feature is displayed', async ({ page }) => {
+    await page.getByRole('link', { name: 'Settings' }).click();
+    expect(page.getByRole('heading', { name: 'Overview' })).toBeVisible();
```
Comment (innerdvations) [inline e2e/tests/settings/smoke-test.spec.ts:23]: It looks like this is failing on github, we probably need the headerVisible fix here too

**Review by innerdvations — APPROVED** [review]

File: e2e/tests/settings/smoke-test.spec.ts (hunk 1 above)
Comment (innerdvations) [inline e2e/tests/settings/smoke-test.spec.ts:23]:
```suggestion
    await expect(page.getByRole('heading', { name: 'Overview' })).toBeVisible();
```

File: e2e/tests/settings/smoke-test.spec.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { test, expect, Page } from '@playwright/test';
+// eslint-disable-next-line import/extensions
+import { resetDatabaseAndImportDataFromPath } from '../../scripts/dts-import';
+import { login } from '../../utils/login';
+import { describeOnCondition } from '../../utils/shared';
+
+const edition = process.env.STRAPI_DISABLE_EE === 'true' ? 'CE' : 'EE';
+
+const headerVisible = async (page: Page, name: string) => {
+  await page.waitForSelector(`:is(h1, h2, h3, h4, h5, h6):has-text("${name}")`);
```
Comment (joshuaellis) [inline e2e/tests/settings/smoke-test.spec.ts:10]: I wonder if we need this, should we just use `getByRole` and leave it?

File: e2e/tests/settings/smoke-test.spec.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+};
+
+test.describe('Settings', () => {
+  test.beforeEach(async ({ page }) => {
+    await resetDatabaseAndImportDataFromPath('./e2e/data/with-admin.tar');
+    await page.goto('/admin');
+    await login({ page });
+  });
+
+  test('every expected feature is displayed', async ({ page }) => {
```
Comment (joshuaellis) [inline e2e/tests/settings/smoke-test.spec.ts:21]: we should check that the EE features are still displayed because we have their purchase page there

**Comment by innerdvations** [comment]
@MarionLemaire I improved the utility functions a bit and pushed the changes we did this afternoon, they should all be passing now!

**Review by Convly — COMMENTED** [review]
@innerdvations / @MarionLemaire what's the status on this one? Do we want to push it through the finish line before GA?

**Comment by innerdvations** [comment]
We'll update it and see if we can get it passing soon, otherwise I'll close :)

**Review by Convly — APPROVED** [review]

**Pull request merged**